pkg/kwok/controllers/pod_controller.go

```diff
@@ -475,6 +478,19 @@ func (c *PodController) configurePod(pod *corev1.Pod, template string) ([]byte,
 		}
 		pod.Status.PodIP = ips[0]
 	}
+	has, node := c.nodeGetFunc(pod.Spec.NodeName)
```

Comment:
Update the NodeIP function in the template

https://github.com/kubernetes-sigs/kwok/blob/8e2871ed268e8efbd3934eeb5432a9afe6844ac2/pkg/kwok/controllers/pod_controller.go#L126-L128

``` golang
"NodeIP": func(nodeName string) string {
	node, has := c.nodeGetFunc(nodeName)
	if has && len(node.HostIPs) > 0 {
		return node.HostIPs[0]
	}
	return n.nodeIP
},
```

and modify https://github.com/kubernetes-sigs/kwok/blob/8e2871ed268e8efbd3934eeb5432a9afe6844ac2/stages/pod-fast.yaml#L60-L61

``` template
      hostIP: {{ with .hostIP -}} 
        {{- . -}}
      {{- else -}}
        {{- NodeIP .metadata.name -}}
      {{- end }}

      podIP: {{ with .podIP -}}
        {{- . -}}
      {{- else -}}
        {{- if .spec.hostNetwork -}}
          {{- NodeIP .metadata.name -}}
        {{- else -}}
          {{- PodIP -}}
        {{- end -}}
      {{- end }}
```

This will be more flexible and versatile.

Reply:
Thank you. Added in the amend change.

Reply:
I changed from `.metadata.name` to `.spec.nodeName`.